internal/operands/metrics/resources.go

```diff
@@ -139,15 +149,123 @@ func getRecordRules() []promv1.Rule {
 	return recordRules
 }
 
+func newMonitoringRbacRole(namespace string) *rbac.Role {
+	return &rbac.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusRole,
+			Namespace: namespace,
+		},
+		Rules: []rbac.PolicyRule{{
+			APIGroups: []string{""},
+			Resources: []string{"services", "endpoints", "pods"},
+			Verbs:     []string{"get", "list", "watch"},
+		}},
+	}
+}
+
+func newMonitoringRbacRoleBinding(namespace string) *rbac.RoleBinding {
+	return &rbac.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusRole,
+			Namespace: namespace,
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      prometheusServiceAccountName,
+				Namespace: MonitorNamespace,
+			},
+		},
+		RoleRef: rbac.RoleRef{
+			Kind:     "Role",
+			Name:     prometheusRole,
+			APIGroup: rbac.GroupName,
+		},
+	}
+}
+
+func newServiceMonitorCR(namespace string) *promv1.ServiceMonitor {
+	return &promv1.ServiceMonitor{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
`TypeMeta`doesn't need to be specified. It is added automatically by the client.

Reply:
done.